app/Http/Controllers/AdministrationProductController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ProductRequest;
+use App\Models\Product;
+use App\Service\ProductService;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\RedirectResponse;
+
+class AdministrationProductController extends Controller
+{
+    private ProductService $productService;
+
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
+
+    public function store(ProductRequest $request): RedirectResponse
+    {
+        if ($request->validated()) {
+            $input = $this->getFieldsFromForm($request);
+
+            $this->productService->save($input);
+
+            return redirect()->to("/products");
+        } else {
+            return redirect()->back();
+        }
+    }
+
+    public function destroy(Product $product): RedirectResponse
+    {
+        $this->productService->destroy($product->id);
+
+        return redirect()->back();
+    }
+
+    public function edit(Product $product): Factory|View|Application
+    {
+        return view("/productEdit", ['product' => $product]);
+    }
+
+    public function update(ProductRequest $request, Product $product): RedirectResponse
+    {
+        $input = $this->getFieldsFromForm($request);
+
+        $this->productService->update($product->id, $input);
```

Comment:
```
$this->productService->update($product->id, [
'name' => $request->name,
             'manufacture' => $request->manufacture,
             'description' => $request->description,
             'releaseDate' => $request->releaseDate,
             'cost' => $request->cost
]);
```

instead of

```
private function getFieldsFromForm(ProductRequest $request): array
     {
         return [
             'name' => $request->name,
             'manufacture' => $request->manufacture,
             'description' => $request->description,
             'releaseDate' => $request->releaseDate,
             'cost' => $request->cost
         ];
     }

Reply:
i use this method in two places? why i gotta replace it?

Reply:
If you want to pass all the data at once, you can pass the request array to your update. otherwise there is no point in holding such methods.

Reply:
```
$this->productService->update($product->id, $request->all());
```

Reply:
Fixed